src/ast/source-range.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export interface SourceRange {
+  /* The resolved path to the file. */
+  file: string;
```

Comment:
I think the way we're doing resolution could mean that we don't need that rule for resolved objects, just for the things being cached. But SourceRange applies to more than resolved things, so...